pkg/multiboot/info.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Multiboot info as defined in
+// https://www.gnu.org/software/grub/manual/multiboot/multiboot.html#Boot-information-format
+package multiboot
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/u-root/u-root/pkg/ubinary"
+)
+
+var sizeofInfo = uint32(binary.Size(Info{}))
+
+const (
+	flagInfoMemory = 1 << iota
+	flagInfoBootDev
+	flagInfoCmdLine
+	flagInfoMods
+	flagInfoAoutSyms
+	flagInfoElfSHDR
+	flagInfoMemMap
+	flagInfoDriveInfo
+	flagInfoConfigTable
+	flagInfoBootLoaderName
+	flagInfoAPMTable
+	flagInfoVideoInfo
+	flagInfoFrameBuffer
+)
+
+// A Info represents a Multiboot v1 information structure.
+type Info struct {
```

Comment:
can you annotate each field's meaning and whether we've implemented putting stuff in it or not?

Reply:
ping2

Reply:
I'm not sure if we want to annotate each field. That will be just a copy of spec. In standard library such structures are not always annotated (e.g [elf header](https://golang.org/src/debug/elf/file.go?s=751:953#L28).
Still, I can mark fields which are ignored for now.